mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/quota/CassandraQuotaCurrentValueDaoTest.java

```diff
@@ -0,0 +1,136 @@
+
+    @BeforeEach
+    private void setup() {
+        cassandraQuotaCurrentValueDao = GuiceUtils.testInjector(cassandraCluster.getCassandraCluster()).getInstance(CassandraQuotaCurrentValueDao.class);
+        resetCounterToZero();
+    }
+
+    private void resetCounterToZero() {
+        cassandraQuotaCurrentValueDao.increase(QUOTA_KEY, 0).block();
+        QuotaCurrentValue quotaCurrentValue = cassandraQuotaCurrentValueDao.getQuotaCurrentValue(QUOTA_KEY).block();
+        cassandraQuotaCurrentValueDao.decrease(QUOTA_KEY, quotaCurrentValue.getCurrentValue()).block();
+        QuotaCurrentValue actual = cassandraQuotaCurrentValueDao.getQuotaCurrentValue(QUOTA_KEY).block();
+        assertThat(actual.getCurrentValue()).isEqualTo(0l);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldCreateNewRowSuccessfully() {
+        QuotaCurrentValue expected = QuotaCurrentValue.builder().quotaComponent(QUOTA_KEY.getQuotaComponent())
+            .identifier(QUOTA_KEY.getIdentifier()).quotaType(QUOTA_KEY.getQuotaType()).currentValue(100l).build();
+        cassandraQuotaCurrentValueDao.increase(QUOTA_KEY, 100l).block();
+
+        QuotaCurrentValue actual = cassandraQuotaCurrentValueDao.getQuotaCurrentValue(QUOTA_KEY).block();
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldCreateNewRowSuccessfullyWhenIncreaseAmountIsZero() {
+        QuotaCurrentValue expected = QuotaCurrentValue.builder().quotaComponent(QUOTA_KEY.getQuotaComponent())
+            .identifier(QUOTA_KEY.getIdentifier()).quotaType(QUOTA_KEY.getQuotaType()).currentValue(0l).build();
+        cassandraQuotaCurrentValueDao.increase(QUOTA_KEY, 0l).block();
+
+        QuotaCurrentValue actual = cassandraQuotaCurrentValueDao.getQuotaCurrentValue(QUOTA_KEY).block();
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldIncreaseValueSuccessfully() {
+        QuotaCurrentValue expected = QuotaCurrentValue.builder().quotaComponent(QUOTA_KEY.getQuotaComponent())
+            .identifier(QUOTA_KEY.getIdentifier()).quotaType(QUOTA_KEY.getQuotaType()).currentValue(200l).build();
+        cassandraQuotaCurrentValueDao.increase(QUOTA_KEY, 100l).block();
+        cassandraQuotaCurrentValueDao.increase(QUOTA_KEY, 100l).block();
+
+        QuotaCurrentValue actual = cassandraQuotaCurrentValueDao.getQuotaCurrentValue(QUOTA_KEY).block();
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void increaseQuotaCurrentValueShouldDecreaseValueSuccessfullyWhenIncreaseAmountIsNegative() {
+        QuotaCurrentValue expected = QuotaCurrentValue.builder().quotaComponent(QUOTA_KEY.getQuotaComponent())
+            .identifier(QUOTA_KEY.getIdentifier()).quotaType(QUOTA_KEY.getQuotaType()).currentValue(100l).build();
+        cassandraQuotaCurrentValueDao.increase(QUOTA_KEY, 200l).block();
+        cassandraQuotaCurrentValueDao.increase(QUOTA_KEY, -100l).block();
```

Comment:
Is that supposed to be possible? Should use decrease instead of increase with negative? Should we have a precondition check on increase and decrease operations that it should be strictly positive numbers?

Reply:
I think "increase -100" means "+ (-100)". It makes sense :))

Reply:
the meaning of increase is really increase... not talking about maths here.

But i'm ok with it, acceptable